doc/source/whatsnew/v1.3.0.rst

```diff
@@ -110,50 +110,42 @@ both XPath 1.0 and XSLT 1.0 is available. (:issue:`27554`)
+We provided some focused development on :class:`.Styler`. See also the `Styler documentation <../user_guide/style.ipynb>`_
+which has been revised and improved (:issue:`39720`, :issue:`39317`, :issue:`40493`).
 
-:meth:`.Styler.set_tooltips` is a new method that allows adding on hover tooltips to
-enhance interactive displays (:issue:`35643`). :meth:`.Styler.set_td_classes`, which was recently
-introduced in v1.2.0 (:issue:`36159`) to allow adding specific CSS classes to data cells, has
-been made as performant as :meth:`.Styler.apply` and :meth:`.Styler.applymap` (:issue:`40453`),
-if not more performant in some cases. The overall performance of HTML
-render times has been considerably improved to
-match :meth:`DataFrame.to_html` (:issue:`39952` :issue:`37792` :issue:`40425`).
-
-The :meth:`.Styler.format` has had upgrades to easily format missing data,
-precision, and perform HTML escaping (:issue:`40437` :issue:`40134`). There have been numerous other bug fixes to
-properly format HTML and eliminate some inconsistencies (:issue:`39942` :issue:`40356` :issue:`39807` :issue:`39889` :issue:`39627`)
-
-:class:`.Styler` has also been compatible with non-unique index or columns, at least for as many features as are fully compatible, others made only partially compatible (:issue:`41269`).
-One also has greater control of the display through separate sparsification of the index or columns, using the new 'styler' options context (:issue:`41142`).
-Render trimming has also been added for large numbers of data elements to avoid browser overload (:issue:`40712`).
-
-We have added an extension to allow LaTeX styling as an alternative to CSS styling and a method :meth:`.Styler.to_latex`
-which renders the necessary LaTeX format including built-up styles (:issue:`21673`, :issue:`41659`). An additional file io function :meth:`Styler.to_html` has been added for convenience (:issue:`40312`).
-
-Documentation has also seen major revisions in light of new features (:issue:`39720` :issue:`39317` :issue:`40493`)
+ - The method :meth:`.Styler.set_table_styles` can now accept more natural CSS language for arguments, such as ``'color:red;'`` instead of ``[('color', 'red')]`` (:issue:`39563`)
+ - The methods :meth:`.Styler.highlight_null`, :meth:`.Styler.highlight_min`, and :meth:`.Styler.highlight_max` now allow custom CSS highlighting instead of the default background coloring (:issue:`40242`)
+ - :meth:`.Styler.set_tooltips` allows on hover tooltips to be added to styled HTML dataframes (:issue:`35643`, :issue:`21266`, :issue:`39317`, :issue:`39708`, :issue:`40284`)
+- :meth:`.Styler.apply` now accepts functions that return an ``ndarray`` when ``axis=None``, making it now consistent with the ``axis=0`` and ``axis=1`` behavior (:issue:`39359`)
+- When incorrectly formatted CSS is given via :meth:`.Styler.apply` or :meth:`.Styler.applymap`, an error is now raised upon rendering (:issue:`39660`)
```

Comment:
@attack68 - I don't see where a raise was added in #39660. Am I just missing it?

I'd like to verify that apply nor applymap will themselves raise, but rather after using them when the Styler renders is were it raises.

Reply:
- `Styler.apply/applymap` necessarily calls `_update_ctx`, **upon render.**
 - `_update_ctx` was changed to call `_maybe_convert_css_to_tuples`. 
 - `_maybe_convert_css_to_tuples` raises on bad CSS.

When `apply/applymap` are called all they do is register themselves in a TODO array to be actually performed later.

Reply:
Thanks!